services/repository/files/content.go

```diff
@@ -101,6 +101,22 @@ func GetContentsOrList(ctx context.Context, repo *repo_model.Repository, treePat
 	return fileList, nil
 }
 
+// GetContentTypeFromTreeEntry check what content is behind it
+func GetContentTypeFromTreeEntry(entry *git.TreeEntry) ContentType {
```

Comment:
Rename to `GetObjectTypeFromTreeEntry`

Reply:
let me check

Reply:
![image](https://user-images.githubusercontent.com/24977596/180272716-70788070-6c3e-4fd6-a67b-dbe0d015bf6d.png)

it's handled as if it's a file link

Reply:
tldr: git has no concept of "directorys" .... so why would they have one for directory-links

Reply:
gitea web ui does handle it similar: https://try.gitea.io/silverwind/symlink-test/src/branch/master/directory-symlink